### ISSUE TITLE
When appropriate, estimators should save `score_` attribute during `fit()`

### ISSUE THREAD

**Issue by rmcgibbo** [msmbuilder/msmbuilder#400]
Currently, osprey computes the distribution of training set _and_ test set scores, by calling `score()` with both datasets. This is inefficient for the training set, since we often know the training set score anyways, without another pass over the data. E.g. for tICA or MSMs, just sum the eigenvalues.

We could save a `score_` attribute on the instance during fit() which osprey could use to avoid this duplication. Some care needs to be taken because for other classification-type problems, you can use a custom scoring function (e.g. entropy loss, or ROC AUC, or whatever), and the estimator's default `score()` function is never called, so the `score_` attrribute would not be the correct value to return. But when the scoring function is the estimator's native `score()`, this seems like a simple win.
